Sophia.md

```diff
@@ -140,96 +70,59 @@ Sophia has the following types:
-
-
-This let-binding doesn't contain any written type:
-
-```ocaml
-let score = 10
+let message = "Hello" // message has value "Hello"
+let message = "Bye" // message has value "Bye"
+message = "Hola" // Error
 ```
 
-Sophia knows that score is an int, judging by the value 10. This is called inference.
+#### Type annotations and inference
+You may have noticed in the previous examples that we haven't given types for the variables. Sophia will infer the value where this makes sense. However sometimes you may want to tell the compiler what the type is, which you do using this syntax
 
-But types can also be explicitly written down by choice:
-
-```ocaml
+```
 let score: int = 10
 ```
-
-You can also wrap any expression in parentheses and annotate it:
-```ocaml
-let myInt = 5
-let myInt: int = 5
-let myInt = (5: int) + (4: int)
-let add = (x: int, y: int) : int => x + y
-// Here add is a function which takes two int as input and returns an int
-// More on function declarations later.
+you can give a type annotation anywhere you introduce a literal or a variable
+```
+let myInt = (5: int) + (10: int)
+let add = (x: int, y: int) : int => x + y  // this is a function definition 
 ```
 
-Declaring types is occasionally necessary with more complicated types, but redundant with simple types.
-
-Types can accept parameters, akin to generics in other languages. It's as if a type is a function that takes in arguments and returns a new type! The parameters need to start with a `'` sign (single quotation mark).
-
-The use-case of a parameterized type is to kill duplications. Before:
+You will only normally declare types with more complicated code.
 
-```ocaml
-/* this is a tuple of 3 items, explained next */
+Types can accept parameters, which are indicated by a leading `'`. Parameterised types are useful when you want to avoid duplication, like this
+##### before
+```
 type intCoordinates = (int, int, int);
 type floatCoordinates = (float, float, float);
 
 let buddy: intCoordinates = (10, 20, 20);
 ```
-
-After:
-
+##### after
 ```
 type coordinates('a) = ('a, 'a, 'a)
-
-/* apply the coordinates "type function" and return the type (int, int, int) */
-type intCoordinatesAlias = coordinates(int)
-
-let buddy: intCoordinatesAlias = (10, 20, 20)
-
-/* or, more commonly, write it inline */
-let buddy: coordinates(float) = (10.5, 20.5, 20.5)
 ```
-The leading apostrophe indicates that `'a` is a type variables: It accept any types. But the variable need to be same in all three fields of the tuple.
-
-In practice, types are inferred for you. So the more concise version of the above example would be nothing but:
-
-```ocaml
-let buddy = (10, 20, 20);
+now you can instantiate the `coordinates` type by giving its type argument
+```
+let buddy: coordinates(float) = (10,20,30)
+```
+type inference means that you can omit the type argument, where it is obvious to the compiler:
+```
+let buddy: coordinates = (10.0, 20.0, 30.0)
```

Comment:
This is not true. You can't omit the argument to `coordinates`. You can write one of
```ocaml
let buddy: coordinates(float) = (10.0, 20.0, 30.0)
let buddy: coordinates(_) = (10.0, 20.0, 30.0)
let buddy: _ = (10.0, 20.0, 30.0)
let buddy = (10.0, 20.0, 30.0)
```